app/src/main/java/com/fabulouszanna/pokedex/model/pokemontypes/PokemonWeaknesses.kt

```diff
@@ -0,0 +1,66 @@
+package com.fabulouszanna.pokedex.model.pokemontypes
+
+import com.fabulouszanna.pokedex.model.PokemonModel
+
+class PokemonWeaknesses(pokemon: PokemonModel) {
+    private val type1 = pokemon.type1
+    private val type2 = pokemon.type2
+
+    private fun getTypeObject(type: String): PokemonType = when (type) {
+        "Bug" -> PokemonType.Bug
+        "Dark" -> PokemonType.Dark
+        "Dragon" -> PokemonType.Dragon
+        "Electric" -> PokemonType.Electric
+        "Fairy" -> PokemonType.Fairy
+        "Fighting" -> PokemonType.Fighting
+        "Fire" -> PokemonType.Fire
+        "Flying" -> PokemonType.Flying
+        "Ghost" -> PokemonType.Ghost
+        "Grass" -> PokemonType.Grass
+        "Ground" -> PokemonType.Ground
+        "Ice" -> PokemonType.Ice
+        "Normal" -> PokemonType.Normal
+        "Poison" -> PokemonType.Poison
+        "Psychic" -> PokemonType.Psychic
+        "Rock" -> PokemonType.Rock
+        "Steel" -> PokemonType.Steel
+        "Water" -> PokemonType.Water
+        else -> throw IllegalArgumentException("Type $type does not exist")
+    }
+
+    private val weaknesses =
+        type2?.let { getTypeObject(type1).weaknesses + getTypeObject(type2).weaknesses }
```

Comment:
if `pokemon.type1` and `pokemon.type2` were `PokemonType` and not a String, this function could look like this

```
val weaknesses = type2?.let { type1?.weakenesses.orEmpty() + it.weakenesses } ?: type1?.weaknesses.orEmpty()
```

Reply:
Yes, this would make code look a lot more clean. However I really don't know how to implement it at the moment